### PR TITLE
Tidied up the interface to be consistent and easier to interact. 

### PR DESCRIPTION
**Pull request by AliMickey**
- Added a semicolon after "weeks" to make it consistent with other variables. 

- Added extra spacing around cells to enable easier interaction and make it look nicer. 

- Fixed spelling of "re-imported"

- Added dashes to improve readability. 

- Centered all graphs on the page.

- Added an 'All' option to the week selection.

### CODE REVIEW

**Comment by AliMickey** [comment]
Updated the versioning, just a question since its my first time. 

Would it not be better to have a versioning system of 4.X.X with 5, 6, 7 being major changes similar to how other projects are done?

File: build.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 name: "jellyfin-plugin-playbackreporting"
 guid: "5c534381-91a3-43cb-907a-35aa02eb9d2c"
-version: "4" # Please increment with each pull request
+version: "5" # Please increment with each pull request
 jellyfin_version: "10.4.0" # The earliest binary-compatible version
```
Comment (JustAMan) [inline build.yaml:5]: did you change .net core version used in the build? 'cause if you didn't it won't work with 10.5+, and if you did you have to change this version as well

**Comment by JustAMan** [comment]
> Would it not be better to have a versioning system of 4.X.X with 5, 6, 7 being major changes similar to how other projects are done?

We do so for main project, but we deemed plugins releases to be rather simple and not very frequent so decided to go with a very simple versioning scheme.

File: build.yaml (hunk 1 above)
Comment (AliMickey) [reply build.yaml:5]: sorry i've never done this before, could you explain what you mean please?

the only things i changed were the html pages so nothing major, am i supposed to update the .net core version somehow? I just used visual studio code. 

I built it for testing using "dotnet publish --configuration Release --output bin" 

Thanks

Comment (JustAMan) [reply build.yaml:5]: This line should probably be updated: https://github.com/jellyfin/jellyfin-plugin-playbackreporting/blob/master/Jellyfin.Plugin.PlaybackReporting/Jellyfin.Plugin.PlaybackReporting.csproj#L3

paging @crobibero for help

Comment (AliMickey) [reply build.yaml]: i had a go please check if correct. thanks

File: Jellyfin.Plugin.PlaybackReporting/Pages/breakdown_report.js [hunk 2]
```diff
@@ -265,8 +265,12 @@ define(['libraryMenu'], function (libraryMenu) {
                 process_click();
 
                 function process_click() {
-                    var days = parseInt(weeks.value) * 7;
-
+                    //Need to implement a better method to select all data instead of 18250 days = 50 years. 
+                    var days = 18250;
+                    if (parseInt(weeks.value) != -1){
+                        var days = parseInt(weeks.value) * 7;
```
Comment (dmitrylyzo) [inline Jellyfin.Plugin.PlaybackReporting/Pages/breakdown_report.js:271]:
```suggestion
                        days = parseInt(weeks.value) * 7;
```

File: Jellyfin.Plugin.PlaybackReporting/Pages/duration_histogram_report.js [hunk 3]
```diff
@@ -200,8 +200,12 @@ define(['libraryMenu'], function (libraryMenu) {
                                 filter.push(filter_name);
                             }
                         }
-
-                        var days = parseInt(weeks.value) * 7;
+                        
+                        //Need to implement a better method to select all data instead of 18250 days = 50 years. 
+                        var days = 18250;
+                        if (parseInt(weeks.value) != -1){
+                            var days = parseInt(weeks.value) * 7;
```
Comment (dmitrylyzo) [inline Jellyfin.Plugin.PlaybackReporting/Pages/duration_histogram_report.js:207]:
```suggestion
                            days = parseInt(weeks.value) * 7;
```

File: Jellyfin.Plugin.PlaybackReporting/Pages/hourly_usage_report.js [hunk 4]
```diff
@@ -402,8 +402,11 @@ define(['libraryMenu'], function (libraryMenu) {
                                 filter.push(filter_name);
                             }
                         }
-
-                        var days = parseInt(weeks.value) * 7;
+                        //Need to implement a better method to select all data instead of 18250 days = 50 years. 
+                        var days = 18250;
+                        if (parseInt(weeks.value) != -1){
+                            var days = parseInt(weeks.value) * 7;
```
Comment (dmitrylyzo) [inline Jellyfin.Plugin.PlaybackReporting/Pages/hourly_usage_report.js:408]:
```suggestion
                            days = parseInt(weeks.value) * 7;
```

File: Jellyfin.Plugin.PlaybackReporting/Pages/user_playback_report.js [hunk 5]
```diff
@@ -484,7 +484,11 @@ define(['libraryMenu'], function (libraryMenu) {
 
                     var weeks = view.querySelector('#weeks');
                     weeks.addEventListener("change", process_click);
-                    var days = parseInt(weeks.value) * 7;
+                    //Need to implement a better method to select all data instead of 18250 days = 50 years. 
+                    var days = 18250;
+                    if (parseInt(weeks.value) != 0){
+                        var days = parseInt(weeks.value) * 7;
```
Comment (dmitrylyzo) [inline Jellyfin.Plugin.PlaybackReporting/Pages/user_playback_report.js:490]:
```suggestion
                        days = parseInt(weeks.value) * 7;
```

File: Jellyfin.Plugin.PlaybackReporting/Pages/user_report.js [hunk 6]
```diff
@@ -78,12 +78,15 @@ define(['libraryMenu'], function (libraryMenu) {
...
-            var days = parseInt(weeks.value) * 7;
 
             process_click();
 
             function process_click() {
-                var days = parseInt(weeks.value) * 7;
+                //Need to implement a better method to select all data instead of 18250 days = 50 years. 
+                var days = 18250;
+                if (parseInt(weeks.value) != -1){
+                    var days = parseInt(weeks.value) * 7;
```
Comment (dmitrylyzo) [inline Jellyfin.Plugin.PlaybackReporting/Pages/user_report.js:88]:
```suggestion
                    days = parseInt(weeks.value) * 7;
```

File: Jellyfin.Plugin.PlaybackReporting/Pages/user_playback_report.js [hunk 7]
```diff
@@ -507,8 +511,16 @@ define(['libraryMenu'], function (libraryMenu) {
...
-                        days = parseInt(weeks.value) * 7;
+                        
+                        /* Since the graph renders each day, rendering all the data does not make sense for this graph. Maybe figure out a way to combine data into months?
+                        //Need to implement a better method to select all data instead of 18250 days = 50 years. 
+                        var days = 18250;
+                        if (parseInt(weeks.value) != -1){
+                            var days = parseInt(weeks.value) * 7;
+                        }
+                        */
+                       days = parseInt(weeks.value) * 7;
```
Comment (dmitrylyzo) [inline Jellyfin.Plugin.PlaybackReporting/Pages/user_playback_report.js:522]:
```suggestion
                        days = parseInt(weeks.value) * 7;
```

**Comment by JustAMan** [comment]
cc @jellyfin/backend

File: Jellyfin.Plugin.PlaybackReporting/Pages/breakdown_report.js (hunk 2 above)
Comment (dmitrylyzo) [reply Jellyfin.Plugin.PlaybackReporting/Pages/breakdown_report.js:271]: For some reason, this one was not applied.

FYI, you can combine suggested changes with `Add suggestion to batch`.

**Comment by AliMickey** [comment]
@dmitrylyzo i reverted your changes on user_playback_report.js, having an all option doesnt make sense as shown in the comment.

File: Jellyfin.Plugin.PlaybackReporting/Pages/user_playback_report.js [hunk 8]
```diff
@@ -484,7 +484,6 @@ define(['libraryMenu'], function (libraryMenu) {
 
                     var weeks = view.querySelector('#weeks');
                     weeks.addEventListener("change", process_click);
-                    var days = parseInt(weeks.value) * 7;
 
                     var url = "user_usage_stats/PlayActivity?filter=" + filter_names.join(",") + "&days=" + days + "&end_date=" + end_date.value + "&data_type=count&stamp=" + new Date().getTime();
```
Comment (dmitrylyzo) [inline Jellyfin.Plugin.PlaybackReporting/Pages/user_playback_report.js:488]: but here `days` is used :confused:

Comment (AliMickey) [reply Jellyfin.Plugin.PlaybackReporting/Pages/user_playback_report.js:488]: its used only once now. How it used to be originally, no all option just 1-12 until a way to merge weeks into months so it displays better, right not is not ideal for that page.

Comment (dmitrylyzo) [reply Jellyfin.Plugin.PlaybackReporting/Pages/user_playback_report.js:488]: I mean `"&days=" + days + "&` and `var days` is inside `process_click`.
I think you should leave `var days = parseInt(weeks.value) * 7;` here and update `days` in `process_click` with new `weeks.value`.

Comment (AliMickey) [reply Jellyfin.Plugin.PlaybackReporting/Pages/user_playback_report.js]: can you request the change again please.

Comment (dmitrylyzo) [reply Jellyfin.Plugin.PlaybackReporting/Pages/user_playback_report.js]: > can you request the change again please.

Which changes? Now it looks fine to me.
Maybe need to make some styling fixes (like space between `){`) and remove extra empty lines.

Comment (AliMickey) [reply Jellyfin.Plugin.PlaybackReporting/Pages/user_playback_report.js]: i reverted before that my bad. 

thanks.

**Comment by dmitrylyzo** [comment]
As alternative (to minimize `parseInt` count).
```js
var days = parseInt(weeks.value) * 7;

// Need to implement a better method to select all data instead of 18250 days = 50 years. 
if (days < 0) days = 18250;
```
Didn't test.

**Comment by AliMickey** [comment]
> 
> 
> As alternative (to minimize `parseInt` count).
> 
> ```js
> var days = parseInt(weeks.value) * 7;
> 
> // Need to implement a better method to select all data instead of 18250 days = 50 years. 
> if (days < 0) days = 18250;
> ```
> 
> Didn't test.

Done, though i made it == -7 to have it stricter. 

thanks

File: Jellyfin.Plugin.PlaybackReporting/Jellyfin.Plugin.PlaybackReporting.csproj [hunk 9]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFramework>netstandard3.1</TargetFramework>
```
[the remark this answers is not in the document]
Comment (crobibero) [reply Jellyfin.Plugin.PlaybackReporting/Jellyfin.Plugin.PlaybackReporting.csproj:3]: netstandard3.1 isn't out yet :)

[the remark this answers is not in the document]
Comment (AliMickey) [reply Jellyfin.Plugin.PlaybackReporting/Jellyfin.Plugin.PlaybackReporting.csproj:3]: Ah thats why i was confused. Ill get it right next time. 


Thanks

File: Jellyfin.Plugin.PlaybackReporting/Pages/duration_histogram_report.js [hunk 10]
```diff
@@ -201,7 +201,9 @@ define(['libraryMenu'], function (libraryMenu) {
                             }
                         }
 
-                        var days = parseInt(weeks.value) * 7;
+                       var days = parseInt(weeks.value) * 7;
+                       if (days == -7) days = 18250;
```
Comment (crobibero) [inline Jellyfin.Plugin.PlaybackReporting/Pages/duration_histogram_report.js:205]: I know this is from @dmitrylyzo but please document what this magic number is for

File: Jellyfin.Plugin.PlaybackReporting/Pages/hourly_usage_report.js [hunk 11]
```diff
@@ -402,8 +402,9 @@ define(['libraryMenu'], function (libraryMenu) {
                                 filter.push(filter_name);
                             }
                         }
-
                         var days = parseInt(weeks.value) * 7;
+                        if (days == -7) days = 18250;
```
Comment (crobibero) [inline Jellyfin.Plugin.PlaybackReporting/Pages/hourly_usage_report.js:406]: same

File: Jellyfin.Plugin.PlaybackReporting/Pages/user_report.js [hunk 12]
```diff
@@ -78,12 +78,13 @@ define(['libraryMenu'], function (libraryMenu) {
 
             var weeks = view.querySelector('#weeks');
             weeks.addEventListener("change", process_click);
-            var days = parseInt(weeks.value) * 7;
 
             process_click();
 
             function process_click() {
                 var days = parseInt(weeks.value) * 7;
+                if (days == -7) days = 18250;
```
Comment (crobibero) [inline Jellyfin.Plugin.PlaybackReporting/Pages/user_report.js:86]: same

File: Jellyfin.Plugin.PlaybackReporting/Pages/duration_histogram_report.js (hunk 10 above)
Comment (AliMickey) [reply Jellyfin.Plugin.PlaybackReporting/Pages/duration_histogram_report.js:205]: i seem to have removed the existing comment, will go ahead and add them in.

Comment (dmitrylyzo) [reply Jellyfin.Plugin.PlaybackReporting/Pages/duration_histogram_report.js:205]: And fix indent please.

**Comment by dmitrylyzo** [comment]
@crobibero Did you have the same (bottom axis labels) for "unlimited" query?
chart
_I did it in hacky way - set weeks to 2700_

**Comment by crobibero** [comment]
> 
> 
> @crobibero Did you have the same (bottom axis labels) for "unlimited" query?
> chart
> _I did it in hacky way - set weeks to 2700_

Yeah, the charting library will write the entire axis unless it's told to skip (https://www.chartjs.org/docs/latest/axes/cartesian/#tick-configuration)

If we find that the axis is getting out of hand `axis.ticks.autoSkip` should be set.

**Pull request merged**